src/prompt.c

```diff
@@ -96,13 +86,21 @@ static char	*get_directory(t_list *env_list)
 	t_list	*home_env;
 
 	cwd_env = env_find(env_list, "PWD");
-	cwd = ((t_env *) cwd_env->content)->value;
-	home_env = env_find(env_list, "HOME");
-	home = ((t_env *) home_env->content)->value;
-	if (ft_strnstr(cwd, home, ft_strlen(cwd)) != NULL)
-		directory = ft_strjoin("~", cwd + ft_strlen(home));
+	if (cwd_env == NULL)
+		cwd = getcwd(NULL, 0);
 	else
+		cwd = ((t_env *) cwd_env->content)->value;
+	home_env = env_find(env_list, "HOME");
+	if (home_env == NULL)
 		directory = ft_strdup(cwd);
+	else
```

Comment:
Check for HOME being unset and fallback on just skipping the decision about tilda and setting the directory to full path